src/credentials/CHIPCert.h

```diff
@@ -186,16 +186,14 @@ enum
  */
 struct ChipRDN
 {
-    union
+    union mAttrValue
     {
-        uint64_t mChipVal; /**< CHIP specific DN attribute value. */
-        struct
-        {
-            const uint8_t * mValue; /**< Pointer to the DN attribute value. */
-            uint32_t mLen;          /**< DN attribute length. */
-        } mString;                  /**< DN attribute structure when encoded as a string. */
-    } mAttrValue;                   /**< DN attribute value union: string or unsigned integer. */
-    chip::ASN1::OID mAttrOID;       /**< DN attribute CHIP OID. */
+        mAttrValue(){};
```

Comment:
What was the reason for adding a ctor & dtor to this union?

nit: Remove unnecessary semicolons

Reply:
Without explicit ctor & dcor definition for mAttrValue union the build fails with the following error:

`default constructor of 'mAttrValue' is implicitly deleted because variant field 'mString' has a non-trivial default constructor`

Reply:
@mspang Please review.

As I mentioned without constructor the compile fail as Spans has non trivial default constructor.

Reply:
```suggestion
        mAttrValue() {}
```

Reply:
Sorry, I still don't think is correct. If using non POD types with C unions you are supposed to call placement new & delete when changing the active member of the union. Anywhere the span is used it is being used before its lifetime starts because it is not POD and it is never constructed.

Perhaps remove the union? Mixing C unions and non-POD types is a subtle corner of C++ and doesn't seem like a good idea. Span has a ctor to make sure it is always initialized.